tasks/preload.js

```diff
@@ -199,38 +199,42 @@ module.exports = function (grunt) {
 						var fileName = resourceMap[preloadFile].fullPath;
 						var fileContent = grunt.file.read(fileName);
 						var fileExtension = path.extname(fileName);
-						
+
 						var iOriginalSize, iCompressedSize;
 
 						if (options.compress) {
 
 							iOriginalSize = fileContent.length;
 							iPreloadOriginalSize += iOriginalSize;
 
-							switch (fileExtension) {
-							case '.js':
-								// Javascript files are processed by Uglify
-								fileContent = uglify.minify(fileContent, {
-									fromString: true,
-									warnings: grunt.option('verbose') === true,
-									output: {
-										comments: copyrightCommentsPattern
+							try {
+								switch (fileExtension) {
+								case '.js':
+									// Javascript files are processed by Uglify
+									fileContent = uglify.minify(fileContent, {
+										fromString: true,
+										warnings: grunt.option('verbose') === true,
+										output: {
+											comments: copyrightCommentsPattern
+										}
+									}).code;
+									break;
+								case '.json':
+									// JSON is parsed and written to string again to remove unwanted white space
+									fileContent = JSON.stringify(JSON.parse(fileContent));
+									break;
+								case '.xml':
+									// For XML we use the pretty data
+
+									// Do not minify if XML(View) contains an <*:pre> tag because whitespace of HTML <pre> should be preserved (should only happen rarely)
+									if (!xmlHtmlPrePattern.test(fileContent)) {
+										fileContent = pd.xmlmin(fileContent, false);
 									}
-								}).code;
-								break;
-							case '.json':
-								// JSON is parsed and written to string again to remove unwanted white space
-								fileContent = JSON.stringify(JSON.parse(fileContent));
-								break;
-							case '.xml':
-								// For XML we use the pretty data
-
-								// Do not minify if XML(View) contains an <*:pre> tag because whitespace of HTML <pre> should be preserved (should only happen rarely)
-								if (!xmlHtmlPrePattern.test(fileContent)) {
-									fileContent = pd.xmlmin(fileContent, false);
-								}
 
-								break;
+									break;
+								}
+							} catch (e) {
+								grunt.fail.warn('Failed to minify ' + fileName + '. This might be due to a syntax error in the file.');
```

Comment:
In general I am ok with the change. There are just a few remarks:

- The original `error` gets lost.
Please use `grunt.fail.warn(e);` (without any custom message) to provide the actual error message and allow to use the [`--stack`](http://gruntjs.com/using-the-cli#stack) option of grunt to optionally print the stack trace.
- Please use something like "process" instead of "minify" for the log message to be more general. You can just add a `grunt.log.error()` to print the custom log message before failing the task with `grunt.fail.warn(e);`